ui/shared/InputFilter.jsx

```diff
@@ -33,15 +34,32 @@ export default class InputFilter extends React.Component {
     }
   };
 
+  userActionListener = async event => {
+    const { updateFilterText } = this.props;
+    const { input } = this.state;
+
+    if (event.key === 'Enter') {
+      updateFilterText(input);
+    }
+  };
+
   render() {
-    const { disabled, placeholder } = this.props;
+    const { disabled, placeholder, updateOnEnter } = this.props;
     const { input } = this.state;
 
+    const inputProps = {
+      onChange: this.updateInput,
+    };
+
+    if (updateOnEnter) {
+      inputProps.onKeyDown = this.userActionListener;
+    }
+
     return (
       <InputGroup>
         <Input
+          {...inputProps}
```

Comment:
Suggestion: please expand this `{...inputProps}` since this would make the code more explicit.